agent/api/task/service_connect_test.go

```diff
@@ -0,0 +1,261 @@
+		StatsConfig: StatsConfig{
+			Path: scPath,
+		},
+		IngressConfig: scIngressConfig,
+		EgressConfig: EgressConfig{
+			ListenerName: scOutListenerName,
+			VIP: VIP{
+				IPV4CIDR: scIPV4Cidr,
+				IPV6CIDR: scIPV6Cidr,
+			},
+		},
+		DNSConfig: []DNSConfigEntry{
+			{
+				HostName: scHostName,
+				Address:  scAddress,
+			},
+		},
+	}
+}
+
+func TestExtractSCConfigFromACSAttachment(t *testing.T) {
+	tt := []struct {
+		testName      string
+		ingressConfig []IngressConfigEntry
+		testACSTask   *ecsacs.Task
+	}{
+		{
+			testName: "AWSVPC default case",
+			ingressConfig: []IngressConfigEntry{
+				{
+					InterceptPort: aws.Uint16(8080),
+					ListenerName:  testInboundListenerName,
+				},
+			},
+			testACSTask: getTestACSTaskAttachment(awsvpcMode, false),
+		},
+		{
+			testName: "AWSVPC port override case",
+			ingressConfig: []IngressConfigEntry{
+				{
+					ListenerPort: uint16(8080),
+				},
+			},
+			testACSTask: getTestACSTaskAttachment(awsvpcMode, true),
+		},
+		{
+			testName: "Bridge default case",
+			ingressConfig: []IngressConfigEntry{
+				{
+					ListenerPort: uint16(15000),
+				},
+			},
+			testACSTask: getTestACSTaskAttachment(bridgeMode, false),
+		},
+		{
+			testName: "Bridge port override case",
+			ingressConfig: []IngressConfigEntry{
+				{
+					ListenerPort: uint16(8080),
+					HostPort:     aws.Uint16(8080),
+				},
+			},
+			testACSTask: getTestACSTaskAttachment(bridgeMode, true),
+		},
+	}
+
+	for _, tc := range tt {
+		t.Run(tc.testName, func(t *testing.T) {
+			testSCConfig := getTestServiceConnectConfig(testHostName, testAddress, testOutboundListenerName, testIPV4Cidr, testIPV6Cidr, testStatsConfigPath, tc.ingressConfig)
+			scConfig, err := ExtractSCConfigFromACSAttachment(tc.testACSTask)
+			assert.NoError(t, err)
+			assert.Equal(t, testSCConfig, scConfig)
+		})
+	}
+}
+
+func TestExtractSCConfigFromACSAttachment_Error(t *testing.T) {
+	testIngressConfig := fmt.Sprintf("\"interceptPort\":\"%s\",\"listenerName\":\"%s\"", testAwsvpcPortDefault, testInboundListenerName)
+	tt := []struct {
+		testName     string
+		testSCConfig string
+	}{
+		{
+			testName:     "AWSVPC default case with no StatsConfig",
+			testSCConfig: fmt.Sprintf(`"{\"egressConfig\":{\"listenerName\":\"%s\",\"vip\":{\"ipv4Cidr\":\"%s\",\"ipv6Cidr\":\"%s\"}},\"dnsConfig\":[{\"hostname\":\"%s\",\"address\":\"%s\"}],\"ingressConfig\":[{%s}]}"`, testOutboundListenerName, testIPV4Cidr, testIPV6Cidr, testHostName, testAddress, testIngressConfig),
+		},
+		{
+			testName:     "AWSVPC default case with no IngressConfig",
+			testSCConfig: fmt.Sprintf(`"{\"statsConfig\":{\"path\":\"%s\"},\"egressConfig\":{\"listenerName\":\"%s\",\"vip\":{\"ipv4Cidr\":\"%s\",\"ipv6Cidr\":\"%s\"}},\"dnsConfig\":[{\"hostname\":\"%s\",\"address\":\"%s\"}]}"`, testStatsConfigPath, testOutboundListenerName, testIPV4Cidr, testIPV6Cidr, testHostName, testAddress),
+		},
+		{
+			testName:     "AWSVPC default case with no EgressConfig",
```

Comment:
wonder why this fails currently if there's no validation for egress config?

Reply:
These test cases might failed on incorrect validation or test setup I had in the previous revision. Please take a look at commit 8e69368bc5c1d7bd3ab71391103f4b6a69f06054 with updated test cases and validations. Thanks.